src/js/component.js

```diff
@@ -38,14 +38,16 @@ import window from 'global/window';
  * @constructor
  * @extends vjs.CoreObject
  */
-var Component = CoreObject.extend({
-  /**
-   * the constructor function for the class
-   *
-   * @constructor
-   */
-  init: function(player, options, ready){
-    this.player_ = player;
+class Component {
+
+  constructor(player, options, ready){
+
+    // The component might be the player itself and we can't pass `this` to super
+    if (!player && this.play) {
+      this.player_ = player = this;
```

Comment:
Yeah, I can't pass `this` (the player) as the first arg to the component constructor or I get yelled at for using `this` before the super().

We could default to this, but I also think it could be good to throw an error if there's no player arg when other components are created. If a component depends on the player API, it will throw random errors later.